app/src/main/java/com/jjswigut/matters/ui/MatterListFragment.kt

```diff
@@ -4,42 +4,53 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.navigation.Navigation
+import androidx.recyclerview.widget.StaggeredGridLayoutManager
+import com.jjswigut.matters.database.MatterDatabase
 import com.jjswigut.matters.databinding.FragmentMatterListBinding
+import kotlinx.coroutines.launch
 
 
 /**
  * A fragment representing a list of Items.
  */
-class MatterListFragment : Fragment() {
+class MatterListFragment : BaseFragment() {
 
     private var _binding: FragmentMatterListBinding? = null
     private val binding get() = _binding!!
-    private lateinit var listAdapter: MatterListRecyclerViewAdapter
+
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
         _binding = FragmentMatterListBinding.inflate(inflater, container, false)
         val view = binding.root
-        initRecyclerView()
-        addDataSet()
+
+        binding.newFab.setOnClickListener {
+            val action = MatterListFragmentDirections.actionMatterListFragmentToEditMatterFragment()
+            Navigation.findNavController(it).navigate(action)
+        }
 
         return view
     }
 
-    private fun addDataSet() {
-        val data = DataSource.createDataSet()
-        listAdapter.submitList(data)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initRecyclerView()
     }
 
+
     private fun initRecyclerView() {
         binding.recyclerView.apply {
-            layoutManager = LinearLayoutManager(context)
-            listAdapter = MatterListRecyclerViewAdapter()
-            adapter = listAdapter
+            layoutManager = StaggeredGridLayoutManager(3, StaggeredGridLayoutManager.VERTICAL)
+            setHasFixedSize(true)
+            launch {
+                context?.let {
+                    val matters = MatterDatabase.getInstance(it).matterDataBaseDao.getAllMatters()
+                    adapter = MatterListRecyclerViewAdapter(matters)
```

Comment:
kind of a weird gotchya - but we don't want to reinitialize our adapter every time the view is created.

we want our adapter to live as long as our fragment does (not just its view)

if you leave it this way, it will work, but you'll notice any time you navigate away from this fragment and back, you'll lose your spot in your recyclerview. this is kind of undesired behavior (picture you're on the twitter app or something with a really long recyclerview and you click on teh 100th post to read it and then navigate back and now you're all the way back at the top)